### ISSUE TITLE
[Bug] — Data leakage during median imputation

### ISSUE THREAD

**Issue by Thebinary-bard** [abhaycs24/CBSOT_SIP_PROJECT-1-#8]
**Type:** Bug
**OSCI'26 Contribution**

### Issue

Missing values in `Total Charges` are currently filled using the median calculated from the complete dataset:

```python
df['Total Charges'].fillna(df['Total Charges'].median(), inplace=True)
```

This preprocessing step occurs before the train/test split.

As a result, information from samples that will later belong to the test set contributes to the median used during training. This introduces data leakage and can lead to overly optimistic evaluation results.

### Proposed Solution

Move the imputation step into the training pipeline so that the median is calculated only from the training data.

The intended workflow should be:

1. Split the dataset into training and test sets.
2. Fit the imputer using `X_train` only.
3. Transform `X_train` using the fitted imputer.
4. Transform `X_test` using the same fitted imputer.
5. Ensure that no statistics from the test set are used during preprocessing.

A `Pipeline` or `ColumnTransformer` based approach can be considered to keep preprocessing consistent and prevent similar leakage in future changes.

### Expected Outcome

The test set should remain completely unseen during preprocessing and model training, resulting in a more trustworthy evaluation of model performance.

I would like to work on this issue as part of my **OSCI'26 contribution**.

**Comment by rushirathod22** [comment]
Hi @abhaycs24, I would like to work on this issue as part of my OSCI'26 contribution. This would be my first open-source contribution, and I’d be happy to take this up.
Could you please assign this issue to me? I’ll follow the project guidelines and submit a clean PR. Thank you! 🙌
